### DISCUSSION TITLE
How to passing variable from keycloak backend to account theme

### DISCUSSION THREAD

**Discussion by codeoleksii** [keycloak/keycloak#23605] [Q&A]
I use keycloak 22.0.1
For account theme I use my own theme it was created according "Keycloak Man" [example](https://github.com/keycloak/keycloak-quickstarts/tree/latest/extension/extend-account-console)
also I put index.ftl in a my theme root this file from keycloak.v2
And in index.ftl I try to get any my custom variable from back

```
${myCustomVar}
```

But didn't work. 

It works for login theme, when I override
`org.keycloak.forms.login.LoginFormsProviderFactory` and then pass attributes
But for account it didn't work, also I used override

```
org.keycloak.theme.DefaultThemeManagerFactory
org.keycloak.theme.freemarker.DefaultFreeMarkerProviderFactory
org.keycloak.theme.ThemeResourceProviderFactory
```

but nothing is work for me, my overrided classes even didn't run I see it in debugger  
Is it possible for account theme? And if possible what class I should override to pass vars.
Thanks in advance for any help!

**Comment by ssilvert** [comment]
If you are just passing in a static string you can specify it in your `theme.properties` file.  So in `theme.properties`, you would just say,

`myCustomVar=myCustomValue`

Then in `index.ftl`, you would say:
`${properties.myCustomVar}`

If you are doing something more complicated then I think you are on the right track.  You should be able to extend/override the DefaultFreeMarkerProvider and stuff a calculated value into the data map.

**Reply by codeoleksii** [reply]
@ssilvert thank you for you answer. Yes I trying to do more complicated scenario. Seems I do wrong overriding. Can't see my class override in terminal when keycloak starting.  Will try it again
